docs/flow/modin/core/dataframe/base/index.rst

```diff
@@ -1,3 +1,17 @@
+Purpose
+=======
+``BaseDataframe`` serves the purpose of describing and defining the internal dataframe algebra.
+
+It is the core construction element which serves as a middle layer between high-level pandas-imitating API
+and lower-level query compiler. ``BaseDataframe`` is the interface which implementations actually translate the dataframe algebra calls to queries to the underlying compiler.
+
+The purpose of such translation is to reduce the vast amount of public pandas API to something smaller but sufficient.
+Query compiler level could reduce the API even further depending on the implementation.
+
+Base dataframe and axis partitions are the interfaces that must be implemented by any backend that wants to be plugged in Modin.
```

Comment:
I wish we would add a base partition manager and a base partition classes that just declare interfaces in near future.

Reply:
This is out of scope of this PR 🙃

Reply:
Definitely

Reply:
I was not able to leave a suggesting here so replace `backend` to `execution backend`/`execution system` please. We are trying to avoid a single `backend` word since the recent restructure.